google-cloud-firestore/lib/google/cloud/firestore/field_path.rb

```diff
@@ -209,19 +209,15 @@ def hash
 
         protected
 
-        START_FIELD_PATH_CHARS = /\A[a-zA-Z_]/.freeze
         INVALID_FIELD_PATH_CHARS = %r{[~*/\[\]]}.freeze
+        SIMPLE_FIELD_PATH_CHARS = /\A[a-zA-Z_][a-zA-Z_0-9]+\Z/.freeze
```

Comment:
I believe `[a-zA-Z_0-9]+` should be changed to `[_a-zA-Z0-9]*` to match the nodejs-firestore [regex](https://github.com/googleapis/nodejs-firestore/blob/master/dev/src/path.ts#L54).

Reply:
@quartzmo you're totally right!!! I am sorry for my poor cautiousness.

Reply:
@quartzmo I have committed to my repository, and follow the regular expressions of other implementations  except for Ruby good regular expression dialect.